src/annotator/config/config.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var settings = require('./settings');
+var sharedSettings = require('../../shared/settings');
+
+/**
+ * Reads the Hypothesis configuration from the environment.
+ *
+ * @param {Window} window_ - The Window object to read config from.
+ */
+function configFrom(window_) {
+  var config = {};
+
+  config.app = settings.iFrameSrc(window_.document);
+
+  Object.assign(config, sharedSettings.jsonConfigsFrom(window_.document));
+  Object.assign(config, settings.configFuncSettingsFrom(window_));
```

Comment:
Many of the config settings are still not mentioned by name in `config.js`, or anywhere in the config/settings code, they're only mentioned where they're used. Those are the settings that come from these blind `Object.assign`s. This also means that non-existent settings returned by js-hypothesis-configs or hypothesisConfigs (`{foo: 'bar'}`) get into the `config` object. I intend to get rid of these `Object.assign`s later and make everything like `config.app` above.